lib/global-config.js

```diff
@@ -43,80 +46,89 @@ if (process.env.USERPROFILE && HOME !== process.env.USERPROFILE && !fs.existsSyn
   }
 }
 
+try {
+  lock();
+
+  // after loading, we'll dprepend all defaults to `exports.config` in this `try` block
+  exports.config = readJSONSync(globalConfigFile);
+
+  // NB can deprecate with jspm 0.14
+  if (!exports.config.defaultRegistry && exports.config.registry)
+    exports.config.defaultRegistry = exports.config.registry;
+  if (!exports.config.registries && exports.config.endpoints)
+    exports.config.registries = exports.config.endpoints;
+
+  // NB add this to deprecate 0.14
+  // delete exports.config.registry;
+  // delete exports.config.endpoints;
+
+  // config upgrade paths
+  // NB can deprecate with jspm < 10
+  if (exports.config.github) {
+    dprepend(exports.config.registries.github, exports.config.github);
+    delete exports.config.github;
+  }
+
+  // populate default registry configuration
+  dprepend(exports.config, {
+    defaultTranspiler: 'babel',
+    defaultRegistry: 'jspm',
+    strictSSL: true,
+    registries: {
+      github: {
+        handler: 'jspm-github',
+        remote: 'https://github.jspm.io'
+      },
+      npm: {
+        handler: 'jspm-npm',
+        remote: 'https://npm.jspm.io'
+      },
+      jspm: {
+        handler: 'jspm-registry',
+        remote: 'https://registry.jspm.io'
+      }
+    }
+  });
+
+  if (HOME)
+    save();
+}
+catch (e) {
+  if (e.code === 'ELOCKED')
+    throw 'Unable to lock global config file, not overwriting';
+}
+finally {
+  unlock();
```

Comment:
Well the whole idea is to keep a single lock throughout the entire process. The lock is only released on three explicit conditions:

1) The process is killed, the lock is released on the exit event
2) An error is thrown, the finally block unlocks the file
3) The procedure completes successfully

So I wouldn't want to `unlock` the file here. I'll explain a bit more in my last comment.